### PR TITLE
Add es6-promise polyfill to webpack config

### PR DESCRIPTION
**Pull request by gnestor**
Implements https://github.com/jupyter/notebook/issues/1754#issuecomment-246994656

### CODE REVIEW

**Comment by gnestor** [comment]
@blink1073 I see that moment and underscore are listed as both npm and bower dependencies so I figure that they can be removed from bower. However, removing them causes conflicts... How do I resolve conflicts when changing dependencies?

**Comment by blink1073** [comment]
What do you mean by conflicts, merge conflicts?  That should only happen if you weren't basing your PR off of master.

**Comment by gnestor** [comment]
I'm actually not sure. You can see that "All checks have failed" and I'm seeing "This branch has conflicts that must be resolved". I created this branch from master, so I'm assuming that this is related to me changing the npm/bower dependencies?

**Comment by blink1073** [comment]
What I mean is that you probably did not have the latest commit from jupyter/notebook master as the basis of your PR, you probably need to fetch and rebase.

**Comment by gnestor** [comment]
You're right, something weird happened...

**Comment by gnestor** [comment]
Travis is complaining that `The command "pip install -f travis-wheels/wheelhouse file://$PWD#egg=notebook[test]" failed and exited with 1 during .` for every build. What does that mean?

**Comment by takluyver** [comment]
I think the significant error message is this bit:

```
    AssertionError: No files match pattern notebook/static/components/es6-promise/*.js
```

Some files are not landing where the Python packaging machinery expects. The package_data is specified here: https://github.com/jupyter/notebook/blob/179bb24fbf79d153812858126127a91431da3319/setupbase.py#L133

File: setupbase.py [hunk 1]
```diff
@@ -135,7 +135,6 @@ def find_package_data():
         pjoin(components, "bootstrap", "js", "bootstrap.min.js"),
         pjoin(components, "bootstrap-tour", "build", "css", "bootstrap-tour.min.css"),
         pjoin(components, "bootstrap-tour", "build", "js", "bootstrap-tour.min.js"),
-        pjoin(components, "es6-promise", "*.js"),
```
Comment (takluyver) [inline setupbase.py]: I don't know exactly where npm puts these files, but I suspect they should still be included in the tarball somehow...

Comment (gnestor) [inline setupbase.py]: I just pushed a new commit that provides the npm paths to these libs.

**Comment by gnestor** [comment]
How can I resolve this `bower.json` conflict?

**Comment by takluyver** [comment]
It must have changed in master as well - fetch master from github, and then `git rebase master` to start rebasing. This is a bit fiddly if you haven't done it before, but you can probably find info online about it.

**Comment by takluyver** [comment]
Oh, and once you've rebased you'll need to force push (`git push -f`) to update the branch on Github.

File: setupbase.py (hunk 1 above)
Comment (gnestor) [inline setupbase.py]: It's still failing and I think it's because it's looking for these files before `npm install`...

**Comment by gnestor** [comment]
Ok I'm just going to forget about removing duplicate dependencies and maybe create an issue about it...

**Comment by gnestor** [comment]
@blink1073 I think this is ready to be merged...

**Comment by blink1073** [comment]
👍, I don't have push rights in this repo.

**Comment by gnestor** [comment]
@takluyver I think this ready to be merged!

**Pull request merged**